subsys/ipc/ipc_service/lib/Kconfig.icmsg

```diff
@@ -78,3 +78,13 @@ config PBUF
 	  with read/write semantics on top of a memory region shared by the
 	  reader and writer. It optionally embeds cache and memory barrier
 	  management to ensure correct data access.
+
+config PBUF_CACHE_HANDLING
+	bool "Cache handling"
+	depends on PBUF
+	default y
+	help
+	  Packet buffer is intended to be used for interprocessor communication
+	  using shared memory. In such case cache must be handled. However, if
+	  packet buffer consumer and producer is on the same core then cache
+	  management shall not be enabled because it may lead to data corruption.
```

Comment:
I don't like this statement. Unnecessary cache management leads to performance overhead and power waste. But it never should result in data corruption. If it does, we have a bug somewhere.